crates/sui-indexer-alt-consistent-store/src/metrics.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use prometheus::{
+    core::{Collector, Desc},
+    proto::MetricFamily,
+    IntGaugeVec, Opts,
+};
+
+pub struct ColumnFamilyStatsCollector {
+    db: Arc<crate::db::Db>,
+    metrics: ColumnFamilyMetrics,
+    cf_names: Vec<String>,
+}
+
+#[derive(Debug)]
+pub struct ColumnFamilyMetrics {
+    /// Mem tables
+    pub current_size_active_mem_tables: IntGaugeVec,
+    /// Active, unflushed immutable, and pinned memtable bytes.
+    pub size_all_mem_tables: IntGaugeVec,
+
+    /// Block Cache
+    pub block_cache_usage: IntGaugeVec,
+    pub block_cache_pinned_usage: IntGaugeVec,
+
+    /// Table reader memory
+    pub estimate_table_readers_mem: IntGaugeVec,
+
+    /// Pending compaction bytes
+    pub estimate_pending_compaction_bytes: IntGaugeVec,
+
+    /// Other indicators of memory backpressure
+    pub num_level0_files: IntGaugeVec,
+    pub actual_delayed_write_rate: IntGaugeVec,
+    pub is_write_stopped: IntGaugeVec,
+    pub num_immutable_mem_tables: IntGaugeVec,
+    pub mem_table_flush_pending: IntGaugeVec,
+    pub compaction_pending: IntGaugeVec,
+
+    /// Informational
+    pub num_snapshots: IntGaugeVec,
+    pub num_running_compactions: IntGaugeVec,
+    pub num_running_flushes: IntGaugeVec,
+}
+
+impl ColumnFamilyStatsCollector {
+    pub fn new(prefix: Option<&str>, db: Arc<crate::db::Db>, cf_names: Vec<String>) -> Self {
+        let metrics = ColumnFamilyMetrics::new(prefix);
+
+        Self {
+            db,
+            metrics,
+            cf_names,
+        }
+    }
+}
+
+/// Create metrics without registering them
+impl ColumnFamilyMetrics {
+    pub(crate) fn new(prefix: Option<&str>) -> Self {
+        let prefix = prefix.unwrap_or("rocksdb");
+        let name = |n| format!("{prefix}_{n}");
+
+        Self {
+            num_level0_files: IntGaugeVec::new(
```

Comment:
Now that you are using the structured (non-protocol) types, you should also be able to use the registration macros to clean this all up and make it more familiar.

We should also use this pattern in `DbConnectionStatsCollector`, if we can, it's a nice pattern!

Although, noting our discussion in pair programming that the macros take a registry and registers the metric as well as constructing it, which we don't want it to do (we want to construct, but not register, because these metrics are registered as part of registering `ColumnFamilyMetrics` as a whole).